### PR TITLE
support azure webapp, plan and publishing_user

### PR DESCRIPTION
**Pull request by GhalebKhaled**

### CODE REVIEW

File: cloudify_azure/resources/app_service/webapp.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        self.logger.info("Delete web app...")
+        self.client.web_apps.delete(
+            resource_group_name=self.group_name,
+            name=self.app_name)
+
+
+@operation
+def create(ctx, resource_group, name, app_config, **kwargs):
+    if ctx.node.properties.get('use_external_resource', False):
+        ctx.logger.info("Using external resource")
```
Comment (EarthmanT) [inline cloudify_azure/resources/app_service/webapp.py]: We should check that the web app exists, like run a get operation. If it fails or returns None, we should fail.

File: cloudify_azure/resources/app_service/plan.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        self.client.app_service_plans.delete(
+            resource_group_name=self.group_name,
+            name=self.plan_name)
+        self.logger.info("Wait for deleting service plan...")
+
+
+@operation
+def create(ctx, resource_group, name, plan_details, **kwargs):
+    if ctx.node.properties.get('use_external_resource', False):
+        ctx.logger.info("Using external resource")
```
Comment (EarthmanT) [inline cloudify_azure/resources/app_service/plan.py]: Verify that plan actually exists.

File: examples/app_service/app-sample-blueprint.yaml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      python_version: 3.6
+
+  user: &user
+    name: sample-user
+    user_name: sample-user
+    publishing_password: STR0NG-test-Pa$$&
+    location: { get_input: location }
+
+node_templates:
+  user:
```
Comment (EarthmanT) [inline examples/app_service/app-sample-blueprint.yaml]: What will someone do if they want to use two distinct users and apps in a single blueprint?

Comment (GhalebKhaled) [reply examples/app_service/app-sample-blueprint.yaml]: you can define two of them in the blueprint then reference them in the nodes.
but, I think there is a limitation on on the users, seems like you can set one user only...
https://github.com/cloudify-incubator/cloudify-azure-plugin/pull/95/files#diff-b50a20f03cef68af08c85c266ddb99a4R34

it works on the webapp, you can define `app_config1` and `app_config2` for example, and set the values you want, at the end, those will be send as JSON to the SDK

**Pull request merged**